File: 347-top-k-frequent-elements/top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        d={}
        for i in nums:
            if i in d:
                d[i]+=1
            else:
                d[i]=1
        d=dict(sorted(d.items(), key=lambda x: x[1],reverse=True))
        i=0
        a=[]
        for ke,v in d.items():
            if i==k :
                break
            i+=1
            a.append(ke)
            
        return a
        
        

File: 347-top-k-frequent-elements/top-k-frequent-elements.java
class Solution {
    public int[] topKFrequent(int[] nums, int k) {
        HashMap<Integer,Integer> h=new HashMap<>();
        for(int i=0;i<nums.length;i++){
            h.put(nums[i],h.getOrDefault(nums[i],0)+1);
        }
        
        List<Map.Entry<Integer,Integer>> li=new ArrayList<>(h.entrySet());
        li.sort((a,b)->b.getValue().compareTo(a.getValue()));
        int[] res=new int[k];
        int j=0;
        for(Map.Entry<Integer,Integer> entry:li){
            if(k==0){
                break;
            }
            res[j]=entry.getKey();
            j+=1;
            k-=1;
        }
        return res;
        
        
    }
}